src/bech32.cpp

```diff
@@ -189,4 +376,96 @@ std::pair<std::string, data> Decode(const std::string& str) {
     return {hrp, data(values.begin(), values.end() - 6)};
 }
 
+/** Find index of an incorrect character in a Bech32 string. */
+int LocateError(const std::string& str, std::string& error) {
+    int ind = CheckCharacters(str);
+    if (ind != -1) {
+        error = "Invalid character";
+        return ind;
+    }
+    if (str.size() > 90) {
+        error = "String too long";
+        return 90;
+    }
+    size_t pos = str.rfind('1');
+    if (pos == str.npos) {
+        error = "Missing separator";
+        return 0;
+    }
+    if (pos == 0 || pos + 7 > str.size()) {
+        error = "Invalid separator position";
+        return pos;
+    }
+    std::string hrp;
+    for (size_t i = 0; i < pos; ++i) {
+        hrp += LowerCase(str[i]);
+    }
+    data values(str.size() - 1 - pos);
+    for (size_t i = 0; i < str.size() - 1 - pos; ++i) {
+        unsigned char c = str[i + pos + 1];
+        int8_t rev = CHARSET_REV[c];
+        if (rev == -1) {
+            error = "Invalid Base 32 character";
+            return i + pos + 1;
+        }
+        values[i] = rev;
+    }
+
+    uint32_t residue = PolyMod(Cat(ExpandHRP(hrp), values)) ^ 1;
+    if (residue != 0) {
+        int length = str.size() - 1;
+        uint32_t syn = Syndrome(residue);
+        int s0 = syn & 0x3FF;
+        int s1 = (syn >> 10) & 0x3FF;
+        int s2 = syn >> 20;
+        int l_s0 = GF1024_LOG[s0];
```

Comment:
9d91c4a046d990d7f648328dfed982ad38c87ff1

Could add at the top:
```cpp
static_assert(sizeof(GF1024_LOG) / sizeof(GF1024_LOG[0]) == 2014, "GF1024_LOG length should be 1024");
```
Same for `GF1024_EXP`.